app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -907,6 +910,12 @@ class GeneralApplicationMongoRepository(
     replaceAllocationStatus(applicationId, EventProgressStatuses.get(eventType.applicationStatus).failedToAttend)
   }
 
+  override def remove(applicationStatus: Option[String]): Future[Int] = {
+    val query = applicationStatus.map(as => BSONDocument("applicationStatus" -> as)).getOrElse(BSONDocument())
```

Comment:
So, that in fact totally opposite of what we should be doing - the Robert C Martin, books, LSCC, and top consultants advocate use of full naming - i.e. no ambiguity